Source/AtomicJS/Javascript/JSAtomic3D.cpp

```diff
@@ -24,6 +25,23 @@ static int StaticModel_SetMaterialIndex(duk_context* ctx) {
     return 0;
 }
 
+static int CustomGeometry_SetMaterial(duk_context* ctx) {
+
+    unsigned index = (unsigned)duk_require_number(ctx, 0);
+    Material* material = js_to_class_instance<Material>(ctx, 1, 0);
+
+    duk_push_this(ctx);
+
+    // event receiver
+    CustomGeometry* geometry = js_to_class_instance<CustomGeometry>(ctx, -1, 0);
+
+
+    geometry->SetMaterial(index, material);
+
+
+    return 0;
+}
```

Comment:
Instead of a custom JS API implementation, is it possible to use overloads capabilities of binding tool to select which to use:

  /// Set material on all geometries.
    void SetMaterial(Material* material);
    /// Set material on one geometry. Return true if successful.
    bool SetMaterial(unsigned index, Material* material);

As with: https://github.com/AtomicGameEngine/AtomicGameEngine/blob/30b5ea42ecd21443d39792a273770c6ff8560de9/Script/Packages/Atomic/Atomic3D.json#L13

If not, any custom API should be also added to  the typescript decl:

https://github.com/AtomicGameEngine/AtomicGameEngine/blob/30b5ea42ecd21443d39792a273770c6ff8560de9/Script/Packages/Atomic/Scene.json#L28